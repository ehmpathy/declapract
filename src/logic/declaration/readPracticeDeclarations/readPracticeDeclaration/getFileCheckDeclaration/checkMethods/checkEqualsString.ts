import { diff } from 'jest-diff';

import { UnexpectedCodePathError } from '../../../../../UnexpectedCodePathError';

export const checkEqualsString = ({
  declaredContents,
  foundContents,
}: {
  declaredContents: string;
  foundContents: string;
}) => {
  try {
    expect(foundContents).toEqual(declaredContents);
  } catch (error) {
    // if the above check failed, run diff on the string directly to show a better string diff message
    const difference = diff(declaredContents, foundContents, { aAnnotation: 'Expected toEqual' });
    if (!difference)
      throw new UnexpectedCodePathError(
        'expect().toEqual() threw an error, but no difference was detected in the strings',
      );
    throw new Error(difference);
  }
};
