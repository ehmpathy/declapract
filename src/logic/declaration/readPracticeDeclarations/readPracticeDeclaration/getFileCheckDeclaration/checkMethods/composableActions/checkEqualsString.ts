import expect from 'expect';
import { diff } from 'jest-diff';

import { UnexpectedCodePathError } from '../../../../../../UnexpectedCodePathError';

export const checkEqualsString = ({
  declaredContents,
  foundContents,
}: {
  declaredContents: string;
  foundContents: string;
}) => {
  try {
    expect(foundContents).toEqual(declaredContents);
  } catch (error) {
    // if the above check failed, run diff on the string directly to show a better string diff message
    const difference = diff(declaredContents, foundContents, {
      aAnnotation: 'Expected toEqual',
    });
    if (
      !difference ||
      difference === 'Compared values have no visual difference.'
    )
      throw new UnexpectedCodePathError(
        'expect().toEqual() threw an error, but no difference was detected in the strings',
        { errorMessage: error.message },
      );
    throw new Error(difference);
  }
};
