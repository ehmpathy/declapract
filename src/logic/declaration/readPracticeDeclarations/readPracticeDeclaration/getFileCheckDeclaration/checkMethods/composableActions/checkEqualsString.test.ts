import { checkEqualsString } from './checkEqualsString';

describe('checkEqualsString', () => {
  it('should return nothing when the string are equal', () => {
    const result = checkEqualsString({ declaredContents: 'a-string', foundContents: 'a-string' });
    expect(result).not.toBeDefined();
  });
  it('should throw an error with a good looking diff when not equal', () => {
    try {
      checkEqualsString({ declaredContents: 'a-string', foundContents: 'b-string' });
      fail('should not reach here');
    } catch (error) {
      expect(error.message).toContain('toEqual');
      expect(error.message).toMatchSnapshot(); // log example of it
    }
  });
});
