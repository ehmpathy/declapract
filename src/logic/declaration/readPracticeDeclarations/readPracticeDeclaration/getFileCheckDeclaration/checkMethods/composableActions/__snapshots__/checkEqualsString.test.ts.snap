// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkEqualsString should throw an error with a good looking diff when not equal 1`] = `
"[32m- Expected toEqual[39m
[31m+ Received[39m

[32m- a-string[39m
[31m+ b-string[39m"
`;
